File: 441-arranging-coins/solution.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        int left = 0, right = n;
        while (left <= right) {
            long long int k = left + (right - left) / 2;
            long long int curr = k * (k + 1) / 2; // calculate the result
            if (curr == n) return k;  // return the right result
            if (curr < n) left = k + 1; 
            else right = k - 1;
        }
        return right;
    }
};

File: 441-arranging-coins/solution.py
class Solution(object):
    # def arrangeCoins(self, n):
    #     """
    #     :type n: int
    #     :rtype: int
    #     """
    #     count = 1
    #     while True:
    #         n -= count
    #         count += 1
    #         if n < 0:
    #             return count - 2
    #         elif n == 0:
    #             return count - 1

    def arrangeCoins(self, n):
        """
        :type n: int
        :rtype: int
        """
        return (int)((2 * n + 0.25)**0.5 - 0.5)
        